fb/app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class CommentsController < ApplicationController
+  before_action :authenticate_user!
+
+  def create
+    @comment = current_user.comments.build(comment_params)
+    redirect_back(fallback_location: root_path) if @comment.save
```

Comment:
This solution isn't the best either. The form will be locked once the first attempt is blank and since its not re rendering, you can comment till you refresh. which is not good.
**Consider this**

```
def create
    @comment = current_user.comments.build(comment_params)
    if @comment.save
      redirect_to root_path
    else
      @pagy, @posts = pagy_array(Post.all.sort_by(&:created_at).reverse)  // you need this line so as to send the posts on that page back to be rendered.
      flash.now[:alert] = 'no blank comments'  // flash.now is a alternative that makes flash available instantly without a new request cycle.
      render 'posts/index'
    end
  end

```

- instead of using flash, consider returning form errors the way you did for the create post form.

Reply:
Sure, I will fix this one.